### PR TITLE
Release.toml: add migration for hostname-override-source

### PR DESCRIPTION
**Pull request by cbgbt**
**Description of changes:**
Adds the migration introduced in #4118 to `Release.toml`

**Testing done:**
* [x] Migration testing in progress
  * `settings.kubernetes.hostname-override-source` becomes present on upgrade
  * The setting is removed on downgrade

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: sources/Cargo.lock [hunk 1]
```diff
@@ -2655,9 +2655,9 @@ dependencies = [
 
 [[package]]
 name = "time"
-version = "0.3.34"
+version = "0.3.36"
```
Comment (arnaldo2792) [inline sources/Cargo.lock:2658]: Is this change needed by the migration or is this change required to fix another problem?

Comment (cbgbt) [reply sources/Cargo.lock:2658]: The build currently fails without this: https://github.com/bottlerocket-os/bottlerocket/actions/runs/10254923209/job/28370732469

I'm not really sure how this didn't cause previous Actions jobs to fail.

Comment (ytsssun) [reply sources/Cargo.lock:2658]: https://github.com/rust-lang/rust/issues/125319#issuecomment-2120809372

There was an regression introduced in `time`. I was following this above issue.

Comment (cbgbt) [reply sources/Cargo.lock:2658]: The Rust compiler has weaker backwards compatibility guarantees specifically regarding type inference, so this was caused by the compiler update with the new SDK.

But I don't understand why the build in #4122 didn't also fail.

**Review by yeazelm — APPROVED** [review]

**Review by ytsssun — APPROVED** [review]

**Review by arnaldo2792 — APPROVED** [review]
